### DISCUSSION TITLE
How to exclude root directory

### DISCUSSION THREAD

**Discussion by jeffreym2-vfi** [rikyoz/bit7z#106] [Q&A] [answered]
I'm using compressDirectory to compress the files in a directory.  However, I'd like to exclude the root directory from the archive.
For example, if the directory structure is:
Root
Root\File 1
Root\File 2
Root\Dir1
Root\Dir1\File 3

I'd like the archive to contain:
File 1
File 2
Dir1
Dir1\File 3

Instead, I'm getting:
Root
Root\File 1
Root\File 2
Root\Dir1
Root\Dir1\File 3

How would I go about getting the behavior that I want?

**Comment by rikyoz** [comment, marked as answer]
Hi!

To achieve the behavior you want, you might use the [BitCompressor::compressFiles](https://github.com/rikyoz/bit7z/wiki/BitCompressor#void-compressfiles-const-wstring-in_dir-const-wstring-out_archive-bool-recursivetrue-const-wstring-filterl--const) method which takes the folder path as its first argument.

```cpp
try {
    Bit7zLibrary lib{};
    BitCompressor compressor{ lib, BitFormat::SevenZip };
    compressor.compressFiles( L"path/to/Root", L"out.7z", true, L"*" );
} catch ( const BitException& ex ) {
    //std::cerr << ex.what() << std::endl;
}
```

Please note that, by default, it uses `L"*.*"` as a filter for the files to be included in the output archive. Hence, you might want to use `L"*"` instead so that it will also consider files without extensions.
